storage/time-series/influxdb/storer/app/App.js

```diff
@@ -0,0 +1,169 @@
+const {
+  ServiceStateManager,
+  ConfigManager: { getConfig, transformObjectKeys },
+  Logger,
+} = require('@dojot/microservice-sdk');
+
+const camelCase = require('lodash.camelcase');
+const InfluxDB = require('./influx');
+const Kafka = require('./kafka');
+
+const {
+  delete: configDelete, lightship: configLightship,
+} = getConfig('STORER');
+
+
+const serviceState = new ServiceStateManager({
+  lightship: transformObjectKeys(configLightship, camelCase),
+});
+serviceState.registerService('influxdb');
+serviceState.registerService('kafka');
+
+const logger = new Logger('influxdb-storer:App');
+
+/**
+* Wrapper to initialize the service
+*/
+class App {
+  /**
+    * Constructor App
+    * that instantiate all kafka and influxdb classes
+    */
+  constructor() {
+    logger.debug('constructor: instantiate app...');
+    try {
+      this.influxDB = new InfluxDB(serviceState);
+      this.kafka = new Kafka(serviceState);
+    } catch (e) {
+      logger.error('constructor:', e);
+      throw e;
+    }
+  }
+
+  /**
+   * Initialize the kafka and influxdb (including HeathChecker and Shutdown)
+   */
+  async init() {
+    logger.info('init: Initializing the influxdb-storer ...');
+    try {
+      const influxIsReady = await this.influxDB.getInfluxStateInstance().isReady();
+      if (!influxIsReady) {
+        throw new Error('Influxdb is not ready');
+      }
+
+      this.influxDB.createHealthChecker();
+      this.influxDB.registerShutdown();
+
+      this.kafka.createHealthChecker();
+      this.kafka.registerShutdown();
+
+      // initializes kafka consumer
+      await this.kafka.getKafkaConsumerInstance().init();
+
+      // associate by callback kafka consumers and influxdb actions
+      this.initCallbacksTenantToCreateAndDelOrgs();
+      this.initCallbacksDevicesCreateAndDelData();
+    } catch (e) {
+      logger.error('init:', e);
+      throw e;
+    }
+    logger.info('init: ... service initialized.');
+  }
+
+  /**
+   * Associates callbacks with kafka devices events to manipulation of influxdb data
+   */
+  initCallbacksDevicesCreateAndDelData() {
+    // create callback to handle receive data
+    const callbackWriteData = async (tenant, deviceid, timestamp, attrs) => {
+      logger.debug('callbackWriteData: init');
+      logger.debug(`callbackWriteData: tenant=${tenant}`);
+      logger.debug(`callbackWriteData: deviceid=${deviceid}`);
+      logger.debug(`callbackWriteData: timestamp=${timestamp}`);
+      logger.debug(`callbackWriteData: attrs=${JSON.stringify(attrs)}`);
+      try {
+        await this.influxDB
+          .getInfluxDataWriterInstance().write(tenant, deviceid, attrs, timestamp);
```

Comment:
It only starts the service if it is able to communicate with influxdb, but after starting it will not stop consuming kafka even if influxdb is out of service, the service is marked as unhealthy.
Any suggestions?

Reply:
we might unsubscribe from Kafka when it becomes unhealthy, and subscribe again when become healthy. I don't Known if there are side effects. Do you have any other suggestion?

Reply:
When you said unsubscribe, do you think about 'unregister callbacks' or 'finish' kafka consumer?

Reply:
I was thinking about 'unregister callbacks', but keep the connection with Kafka. Stop consuming.